ckanext/datatablesview/blueprint.py

```diff
@@ -59,13 +59,16 @@ def ajax(resource_view_id: str):
     filters = merge_filters(view_filters, user_filters)
 
     datastore_search = get_action(u'datastore_search')
-    unfiltered_response = datastore_search(
-        {}, {
-            u"resource_id": resource_view[u'resource_id'],
-            u"limit": 0,
-            u"filters": view_filters,
-        }
-    )
+    try:
+        unfiltered_response = datastore_search(
+            {}, {
+                u"resource_id": resource_view[u'resource_id'],
+                u"limit": 0,
+                u"filters": view_filters,
+            }
+        )
+    except ObjectNotFound:
+        return json.dumps({'error': 'Object not found'})
```

Comment:
Is this the response datatables expects on errors? Does it render the text in a dialog box or something?

Should probably also set the response status to 404.

Reply:
@wardi This function is a response for a POST and only returns JSON with no status (even for errors).
https://github.com/ckan/ckan/blob/master/ckanext/datatablesview/blueprint.py#L141
I just add statu codes for all responses

Reply:
> Should also catch NotAuthorized here

Added here https://github.com/avdata99/ckan/commit/ff1b9d758d502fbabbe3707a4efd613bbee97a16